### PR TITLE
VASP: updating write magmoms

### PR DESCRIPTION
**Pull request by FeLoch**
Will close the attached issue.

### CODE REVIEW

File: pyiron/vasp/base.py [hunk 1]
```diff
@@ -727,58 +727,63 @@ def write_magmoms(self):
...
-                        "LAMBDA is not specified but it is necessary for non collinear calculations."
-                    )
-                if (
-                    self.spin_constraints
-                    and "RWIGS" not in self.input.incar._dataset["Parameter"]
+            if self.input.incar["ISPIN"] != 1:
+                final_cmd = "   ".join(
+                    [
+                        " ".join([str(spinmom) for spinmom in spin])
+                        if isinstance(spin, list) or isinstance(spin, np.ndarray)
```
Comment (sudarsan-surendralal) [inline pyiron/vasp/base.py:736]: Better to write `if  isinstance(spin, (list, np.ndarray))`

File: pyiron/vasp/base.py [hunk 2]
```diff
@@ -727,58 +727,63 @@ def write_magmoms(self):
...
+                        else str(spin)
+                        for spin in self.structure.get_initial_magnetic_moments()[
+                            self.sorted_indices
+                        ]
+                    ]
+                )
+                s.logger.debug("Magnetic Moments are: {0}".format(final_cmd))
+                if "MAGMOM" not in self.input.incar._dataset["Parameter"]:
+                    self.input.incar["MAGMOM"] = final_cmd
+                if any(
```
Comment (sudarsan-surendralal) [inline pyiron/vasp/base.py:746]: This too can be simplified

```
any([isinstance(spin, (list, np.ndarray)) for spin in self.structure.get_initial_magnetic_moments()])
```

File: pyiron/vasp/base.py (hunk 1 above)
Comment (FeLoch) [reply pyiron/vasp/base.py:736]: , I thought, we have tests for this? The github highlighting is a little bit strange as I only added an outer if statement 😄

Comment (FeLoch) [reply pyiron/vasp/base.py]: done

File: pyiron/vasp/base.py (hunk 2 above)
Comment (FeLoch) [reply pyiron/vasp/base.py:746]: done

File: pyiron/vasp/base.py (hunk 1 above)
Comment (sudarsan-surendralal) [reply pyiron/vasp/base.py]: > 
> 
> , I thought, we have tests for this? The github highlighting is a little bit strange as I only added an outer if statement 😄

Codeacy only highlights pep8 violations. It does not suggest better coding styles unfortunately!

**Comment by FeLoch** [comment]
@sudarsan-surendralal  I added the test, would you like to have a look?

**Pull request merged**